src/rest/Users/addUser.ts

```diff
@@ -82,8 +84,13 @@ export const addUser = async (
 	}
 
 	try {
-		await addUserToDb(userDetails);
-		res.status(StatusCode.OK).json(UserCreatedJSON);
+		const addedUser = await addUserToDb(userDetails);
+		const token = createToken();
+		await editUserInDb(addedUser._id as ObjectId, {
+			token: token,
+			dateTokenCreated: new Date(),
+		});
+		res.status(StatusCode.OK).json({ ...UserCreatedJSON, token: token });
```

Comment:
Why not just add `token` and `dateTokenCreated` to the user object on the initial insertion instead of calling `editUserInDb`? If we're trying to be more efficient, then doing it all in one shot would be most efficient.